weed/shell/command_volume_fsck.go

```diff
@@ -541,19 +543,38 @@ func (c *commandVolumeFsck) oneVolumeFileIdsSubtractFilerFileIds(dataNodeId stri
 		return
 	}
 
+	voluemAddr := pb.NewServerAddressWithGrpcPort(dataNodeId, 0)
 	if err = c.readFilerFileIdFile(volumeId, func(nId types.NeedleId, itemPath util.FullPath) {
+		inUseCount++
+		if *c.verifyNeedle {
+			if v, ok := db.Get(nId); ok && v.Size.IsValid() {
+				newSize := types.Size(16)
+				if v.Size > newSize {
+					v.Size = newSize
+				}
+				if _, err := readSourceNeedleBlob(c.env.option.GrpcDialOption, voluemAddr, volumeId, *v); err != nil {
+					fmt.Fprintf(c.writer, "failed to read file %s NeedleBlob %+v: %+v", itemPath, nId, err)
+					if *c.forcePurging {
+						return
+					}
```

Comment:
not clear what is the logic here.

Reply:
> not clear what is the logic here.

@chrislusf Logic in removing chunks that if their blobs are not available

Reply:
seems the logic should be opposite.

Reply:
> seems the logic should be opposite.

@chrislusf It's a little unclear here. Since the next function is `db.Delete(nId)`
It removes the needle's metadata from the database, then deletes what is left.

That's why
```
if *c.forcePurging {
return
}
```
Ensures that honey melon needles stay in the database and are labeled as orphan.